core/src/ServiceBundle.cs

```diff
@@ -0,0 +1,81 @@
+// 
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+// 
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+// 
+// ------------------------------------------------------------------------------
+
+using Microsoft.Identity.Core.Http;
+using Microsoft.Identity.Core.Instance;
+using Microsoft.Identity.Core.Telemetry;
+
+namespace Microsoft.Identity.Core
+{
+    internal class ServiceBundle : IServiceBundle
+    {
+        internal ServiceBundle(
+            ICoreExceptionFactory coreExceptionFactory = null,
+            IHttpClientFactory httpClientFactory = null,
+            IHttpManager httpManager = null,
+            ITelemetryReceiver telemetryReceiver = null,
+            IValidatedAuthoritiesCache validatedAuthoritiesCache = null,
+            IAadInstanceDiscovery aadInstanceDiscovery = null,
+            bool shouldClearCaches = false)
+        {
+            ExceptionFactory = coreExceptionFactory ?? CoreExceptionFactory.Instance;
+            HttpManager = httpManager ?? new HttpManager(coreExceptionFactory, httpClientFactory);
+            TelemetryManager = new TelemetryManager(telemetryReceiver);
+            ValidatedAuthoritiesCache = new ValidatedAuthoritiesCache(shouldClearCaches);
+            AadInstanceDiscovery = new AadInstanceDiscovery(HttpManager, TelemetryManager, shouldClearCaches);
+            PlatformProxy = PlatformProxyFactory.GetPlatformProxy();
+        }
+
+        /// <inheritdoc />
+        public IHttpManager HttpManager { get; }
+
+        /// <inheritdoc />
+        public ITelemetryManager TelemetryManager { get; }
+
+        /// <inheritdoc />
+        public IValidatedAuthoritiesCache ValidatedAuthoritiesCache { get; }
+
+        /// <inheritdoc />
+        public IAadInstanceDiscovery AadInstanceDiscovery { get; }
+
+        /// <inheritdoc />
+        public ICoreExceptionFactory ExceptionFactory { get; }
+
+        /// <inheritdoc />
+        public IPlatformProxy PlatformProxy { get; }
+
+        public static ServiceBundle CreateForProduction(ITelemetryReceiver telemetryReceiver = null)
+        {
+            return new ServiceBundle(telemetryReceiver: telemetryReceiver);
+        }
+
+        public static ServiceBundle CreateForTest(IHttpManager httpManager, ITelemetryReceiver telemetryReceiver = null)
+        {
+            return new ServiceBundle(httpManager: httpManager, telemetryReceiver: telemetryReceiver, shouldClearCaches: true);
+        }
```

Comment:
based on the current test code I would like that these two have a description on when to be used and what the difference really is. Perhaps instead of test and prod those should be named CreateDefaultBundle() and CreateBundleWithCleanCache() ?

Reply:
I've renamed them to CreateDefault and CreateWithCustomHttpManager since those are the actual usages.